cpp_tests/feature/one_hot_encoder_test.cc

```diff
@@ -122,42 +156,67 @@ TEST_F(OneHotEncoderTest, Mlir) {
 
   test::saveMetaDataModelData(tmp_folder.getAbsolutePath(), params, model_data);
 
-  // TODO: Separate the load process that is triggered only once and the
-  // repeatedly triggered transform process into different scripts.
-  auto mlir_script = R"mlir(
-    func @main(%path: !tfrt.string, %value: i32, %column_index: i32) -> !clink.vector {
-      %model = clink.onehotencoder_load %path
-      %vector = clink.onehotencoder_transform %model, %value, %column_index
-      tfrt.return %vector : !clink.vector
+  const std::string mlir_script = R"mlir(
+    func @load_model(%path: !tfrt.string) -> !clink.model {
+      %model = clink.load.onehotencoder %path
+      tfrt.return %model : !clink.model
+    }
+
+    func @transform_inputs(%model: !clink.model, %value: i32, %column_index: i32) -> !clink.vector {
+        %outputs = clink.transform %model, %value, %column_index : (i32, i32) -> !clink.vector
+        tfrt.return %outputs : !clink.vector
     }
   )mlir";
 
-  llvm::SmallVector<RCReference<AsyncValue>, 4> inputs;
-  inputs.push_back(tfrt::MakeAvailableAsyncValueRef<std::string>(
-      tmp_folder.getAbsolutePath()));
-  inputs.push_back(tfrt::MakeAvailableAsyncValueRef<int32_t>(1));
-  inputs.push_back(tfrt::MakeAvailableAsyncValueRef<int32_t>(0));
-
-  auto results =
-      test::runMlirScript(host_context, mlir_context, mlir_script, inputs);
-  EXPECT_EQ(results.size(), 1);
-  host_context->Await(results);
-  SparseVector &actual_vector = results[0]->get<SparseVector>();
-  SparseVector expected_vector(2);
-  expected_vector.set(1, 1.0);
-  EXPECT_EQ(actual_vector, expected_vector);
+  clink::ClinkRunner::Builder builder;
+  builder.set_mlir_fn_name("load_model")
+      .set_mlir_input(mlir_script)
+      .set_host_context(host_context)
+      .set_mlir_context(mlir_context);
+  auto model_load_runner = builder.Compile();
+
+  llvm::SmallVector<RCReference<AsyncValue>> model_load_inputs{
+      tfrt::MakeAvailableAsyncValueRef<std::string>(
+          tmp_folder.getAbsolutePath())};
+  auto model_load_outputs = model_load_runner.Run(model_load_inputs);
+  host_context->Await(model_load_outputs);
+  auto model_ref = model_load_outputs[0];
+
+  builder.set_mlir_fn_name("transform_inputs");
+  auto model_transform_runner = builder.Compile();
+
+  {
+    llvm::SmallVector<RCReference<AsyncValue>, 4> inputs;
+    inputs.push_back(model_ref);
+    inputs.push_back(MakeAvailableAsyncValueRef<int>(1).CopyRCRef());
```

Comment:
Since we don't expect to use the newly-instantiated for other purpose, should we use `ReleaseRCRef()` instead of `CopyRCRef()`? Similarly for other usages of `ReleaseRCRef()`.

Reply:
I found that this is enough.
```suggestion
    inputs.push_back(MakeAvailableAsyncValueRef<int>(1);
```
Given that their implementation is the same.
```c++
template <typename T>
class AsyncValueRef {
 public:

  // Allow implicit conversion to type-erased RCReference<AsyncValue>
  operator RCReference<AsyncValue>() && { return std::move(value_); }

  // Transfer ownership of one reference on the AsyncValue to the returned
  // RCReference<AsyncValue>.
  RCReference<AsyncValue> ReleaseRCRef() { return std::move(value_); }

 private:
  RCReference<AsyncValue> value_;
};
```
I'll make the change in this way.